agents/scsi/fence_scsi.py

```diff
@@ -324,9 +324,9 @@ def get_clvm_devices(options):
 	"--config 'global { locking_type = 0 } devices { preferred_names = [ \"^/dev/dm\" ] }'"
 	out = run_cmd(options, cmd)
 	if out["rc"]:
-		fail_usage("Failed: Cannot get clvm devices")
+		fail_usage("Failed: Cannot get shared devices")
 	for line in out["out"].split("\n"):
-		if 'c' in line.split(":")[0]:
+		if 'c' in line.split(":")[0] or 's' in line.split(":")[0]:
```

Comment:
This matches the existing pattern so there's nothing wrong with it. To add clarity and avoid redundancy, we could do:
```
        for line in out["out"].split("\n"):
                vg_attr, pv_name = line.split(":")
                if 'c' in vg_attr or 's' in vg_attr:
                        devs.append(pv_name)
```

There's also an edge case in the *existing* code that would generate a false positive: a `'c'` character in the fifth position indicates a contiguous allocation policy, not a clustered volume group. A `'c'` in the sixth position indicates a clustered volume group. From the `vgs(8)` man page:
```
       The vg_attr bits are:

       1  Permissions: (w)riteable, (r)ead-only
       2  Resi(z)eable
       3  E(x)ported
       4  (p)artial: one or more physical volumes belonging to the volume group are missing from the system
       5  Allocation policy: (c)ontiguous, c(l)ing, (n)ormal, (a)nywhere
       6  (c)lustered, (s)hared
```

I'm not 100% sure whether that VG attr bit order is consistent on all supported OSes, but `get_clvm_devices()` is sort of a guess anyway... the user can always specify a devices list if there's a problem with the dynamic check in `get_clvm_devices()`.

If we check only the sixth position (index 5 of the string), then the code could be as follows:
```
        for line in out["out"].split("\n"):
                vg_attr, pv_name = line.split(":")
                if vg_attr[5] in 'cs':
                        devs.append(pv_name)
```

Reply:
We tweaked this a little bit after Arslan tested. I think we settled on:
```
        for line in out["out"].splitlines():
                vg_attr, pv_name = line.strip().split(":")
                if vg_attr[5] in "cs":
                        devs.append(pv_name)
        return devs
```

Reply:
Thank you Reid for recommendations and help in getting it through.